src/vos/vos_ts.h

```diff
@@ -537,4 +538,50 @@ vos_ts_check_rh_conflict(struct vos_ts_set *ts_set, daos_epoch_t write_time)
 	return true;
 }
 
+static inline bool
+vos_ts_check_rh_conflict_type(struct vos_ts_set *ts_set, uint32_t type,
+			      daos_epoch_t write_time)
+{
+	struct vos_ts_entry	*entry;
+	uint32_t		 akey_nr = 0;
+
+	if (ts_set == NULL)
+		return false;
+
+	if (type == VOS_TS_TYPE_AKEY)
+		akey_nr = ts_set->ts_init_count - 3;
```

Comment:
It's better to have some comments to explain the 'ts_init_count' and '2', '3' somewhere.

Reply:
I added a constant